src/evaluators/UnaryExpression.js

```diff
@@ -154,7 +154,11 @@ function evaluateOperation(
   strictCode: boolean,
   realm: Realm
 ): Value {
-  function reportError() {
+  function reportError(value: Value) {
+    if (value.getType() === SymbolValue) {
+      // Symbols never implicitly coerce to primitives.
```

Comment:
You probably want to move this to follow the handleError call. I worried that it might be too strict, but it seems that it will only get here for operations that will always fail on symbols.

Reply:
That's exactly what I want to avoid. This causes problems because the error is "recoverable" and then the runtime value can be a symbol which immediately throws. Even if you don't have any valueOf/toString/toPrimitive overrides in your code. It also causes trouble running test262 in abstract mode which will error compiling instead of passing.

Reply:
I see. Since the operation is known to be an error at compile time, we may as well process it as such. If something handles it, fine, if not it bubbles to the top level and gets a message at that point.

I do think, however, that is highly confusing to do this in a function called reportError. Perhaps just rename reportError to reportPossibleErrorOrDoThrowCompletion.